src/test/java/tdl/record/sourcecode/content/CopyFromDirectorySourceCodeProviderTest.java

```diff
@@ -0,0 +1,51 @@
+package tdl.record.sourcecode.content;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import tdl.record.sourcecode.test.FileTestHelper;
+
+public class CopyFromDirectorySourceCodeProviderTest {
+
+    @Rule
+    public TemporaryFolder folder = new TemporaryFolder();
+
+    @Test
+    public void retrieveAndSaveToShouldHonourGitignoreContent() throws IOException, GitAPIException {
+        File original = folder.newFolder();
+        Path originalPath = original.toPath();
+        File actual = folder.newFolder();
+
+        Git git = Git.init().setDirectory(original).call();
+        FileTestHelper.appendStringToFile(originalPath, "file1.txt", "TEST");
+        FileTestHelper.appendStringToFile(originalPath, "subdir1/file1.txt", "TEST");
+        FileTestHelper.appendStringToFile(originalPath, "file1.bak", "TEST");
+        FileTestHelper.appendStringToFile(originalPath, ".gitignore", "*.bak");
+
+        git.add().addFilepattern(".").call();
```

Comment:
We also need to copy files that are not tracked in Git yet, but not ignored.

Reply:
I've tried to find how to list untracked and committed but not ignored using TreeWalk. No result. So we will need two way, one by using TreeWalk, another one using git status.